### DISCUSSION TITLE
Speed comparison with ROS1 localhost python, C++, C++ nodelets?

### DISCUSSION THREAD

**Discussion by lucasw** [dora-rs/dora#455] [Q&A]
I saw a link to this on https://www.weeklyrobotics.com.  How hard would it be to compare performance to ROS1 localhost?  Is the ros2 benchmarking code in a repo here, could be used as a model for something similar in ros1?

**Comment by haixuanTao** [comment]
Sure thing, i'll put in the repo this weekend, with a docker image to make it dead easy to try out

**Comment by haixuanTao** [comment]
Reproduction is now live here: https://github.com/dora-rs/dora-benchmark/tree/main

just do the following:
```bash
git clone https://github.com/dora-rs/dora-benchmark.git
cd dora-benchmark
```

You should be able to reproduce the result in python fairly easily.

For ROS 1, I haven't really tried, sorry. I guess it should be fairly similar.
